demo/src/main/java/com/example/demo/dao/gamestate/GameStateDao.java

```diff
@@ -0,0 +1,14 @@
+package com.example.demo.dao.gamestate;
+
+import com.example.demo.models.dto.GameState;
+
+import java.util.List;
+
+public interface GameStateDao {
+    GameState getStateById(Long id);
+    GameState getStateByUserId(Long id);
+    GameState saveStateGame(GameState game);
+    List<GameState> getAll();
+    void setGame(Long idGame, Long idUserOne, Long idUserSecond, GameState game);
+    Long generateGameId();
```

Comment:
probably id generation can be hidden, especially considering  the fact that usually databases do it on insert.